src/community/assortativity.jl

```diff
@@ -34,18 +48,34 @@ function assortativity(g::AbstractGraph{T}) where T
     return assortativity_coefficient(g, sjk, sj, sk, sjs, sks, nue)
 end
 
+function assortativity(g::AbstractGraph{T}, attributes::Vector{N}) where {T,N<:Number}
```

Comment:
`T` needs to be `<: Integer`. Can `N` really be `<: Number`? Does this work with complex?

Reply:
Thanks for your comment. `T` is now as requested `<: Integer`. You are right, `N` can not be `Complex`, I thus modified it as `<:Real`.